axelrod/strategies/memorytwo.py

```diff
@@ -1,11 +1,181 @@
+        self.classifier['stochastic'] = any(0 < x < 1 for x in set(sixteen_vector))
+
+    def strategy(self, opponent: Player) -> Action:
+        if len(opponent.history) <= 1:
+            return self._initial
+        # Determine which probability to use
+        p = self._sixteen_vector[(tuple(self.history[-2:]),
+                                  tuple(opponent.history[-2:]))]
+        # Draw a random number in [0, 1] to decide
+        return random_choice(p)
+
+
+class AON2(MemoryTwoPlayer):
+    """
+    AON2 a memory two strategy introduced in [Hilbe2017]_. It belongs to the
+    AONk (all-or-none) family of strategies. These strategies were designed to
+    satisfy the three following properties:
+
+    1. Mutually Cooperative. A strategy is mutually cooperative if there are
+    histories for which the strategy prescribes to cooperate, and if it continues
+    to cooperate after rounds with mutual cooperation (provided the last k actions
+    of the focal player were actually consistent).
+
+    2.Error correcting. A strategy is error correcting after at most k rounds if,
+    after any history, it generally takes a group of players at most k + 1 rounds
+    to re establish mutual cooperation.
+
+    3. Retaliating. A strategy is retaliating for at least k rounds if, after
+    rounds in which the focal player cooperated while at least one coplayer defected,
+    the strategy responds by defecting the following k rounds.
+
+    In essence is a strategy that starts of by cooperating and will cooperate
+    again only after the states (CC, CC), (CD, CD), (DC, DC), (DD, DD).
+
+    Names:
+
+    - AON2: [Hilbe2017]_
+    """
+
+    name = 'AON 2'
+    classifier = {
+        'memory_depth': 2,  # Memory-one Four-Vector
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self, initial: Action = C) -> None:
+        sixteen_vector = (1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1)
+        super().__init__(sixteen_vector)
+        self._initial = initial
+
+
+class DelayedAON1(MemoryTwoPlayer):
+    """
+    Delayed AON1 a memory two strategy also introduced in [Hilbe2017]_ and belongs to the
+    AONk family. Note that AON1 is equivalent to Win Stay Lose Swift.
+
+    In essence is a strategy that starts of by cooperating and will cooperate
+    again only after the states (CC, CC), (CD, CD), (CD, DD), (DD, CD),
+    (DC, DC) and (DD, DD).
+
+    Names:
+
+    - Delayed AON1: [Hilbe2017]_
+    """
+
+    name = 'Delayed AON1'
+    classifier = {
+        'memory_depth': 2,  # Memory-one Four-Vector
+        'stochastic': False,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def __init__(self, initial: Action = C) -> None:
```

Comment:
As these strategies are defined in the literature, do we need `initial` to be an option?